src/libslic3r/GCode/GCodeProcessor.cpp

```diff
@@ -4357,14 +4394,23 @@ void GCodeProcessor::set_travel_acceleration(PrintEstimatedStatistics::ETimeMode
 
 float GCodeProcessor::get_filament_load_time(size_t extruder_id)
 {
-    //BBS: change load time to machine config and all extruder has same value
-    return m_time_processor.extruder_unloaded ? 0.0f : m_time_processor.filament_load_times;
+    if (s_IsBBLPrinter) {
+        // BBS: change load time to machine config and all extruder has same value
+        return m_time_processor.extruder_unloaded ? 0.0f : m_time_processor.filament_load_times[0];
+    } else {
+        return (m_time_processor.filament_load_times.empty() || m_time_processor.extruder_unloaded) ?
+                   0.0f :
+                   ((extruder_id < m_time_processor.filament_load_times.size()) ? m_time_processor.filament_load_times[extruder_id] :
+                                                                                  m_time_processor.filament_load_times.front());
+    }
 }
 
 float GCodeProcessor::get_filament_unload_time(size_t extruder_id)
 {
-    //BBS: change unload time to machine config and all extruder has same value
-    return m_time_processor.extruder_unloaded ? 0.0f : m_time_processor.filament_unload_times;
+    return (m_time_processor.filament_unload_times.empty() || m_time_processor.extruder_unloaded) ?
```

Comment:
This code also works.
Just so that we keep the logic consistent, can we either:
1. Add `if (s_IsBBLPrinter)` logic in this function
2. Or Remove `if (s_IsBBLPrinter)` logic in get_filament_load_time() function?

Reply:
aah sorry that one slipped from my eyes 👍